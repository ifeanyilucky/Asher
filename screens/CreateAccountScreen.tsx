import * as React from "react";
import {
  View,
  Text,
  SafeAreaView,
  TouchableOpacity,
  Image,
  TextInput,
  Platform,
  StyleSheet,
} from "react-native";
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { RootStackParamList } from "../navigation/types";
import { Icon } from "react-native-elements";
import { exo2 } from "../styles/typography";

type Props = NativeStackScreenProps<RootStackParamList, "Recommendations">;

export default function CreateAccountScreen({ navigation }: Props) {
  const [email, setEmail] = React.useState<string>("");
  const [error, setError] = React.useState<string>("");

  // Function to handle form submission
  const handleSubmit = () => {
    if (!email) {
      // Check if email is empty
      setError("Email is required.");
    } else if (!/\S+@\S+\.\S+/.test(email)) {
      // Validate email format using regex
      setError("Email is invalid.");
    } else {
      // Navigate to the "ConfirmPassword" screen if email is valid
      navigation.navigate("ConfirmPassword");
    }
  };

  return (
    <SafeAreaView className="bg-white h-screen flex">
      <View className="mt-2 mb-3 mx-3">
        <TouchableOpacity
          className="p-2 inline-block w-12"
          onPress={() => navigation.goBack()}
          testID="back-button"
        >
          <Icon name="arrow-back-ios" />
        </TouchableOpacity>
      </View>
      <View className="bg-white mx-10 h-screen">
        <View className="flex justify-center items-center my-10">
          <Text
            className="text-4xl font-bold text-black text-center"
            style={exo2.bold}
          >
            CREATE AN ACCOUNT
          </Text>
        </View>

        <View className="flex items-center mb-10">
          <View className="bg-[#B80238] p-3 rounded-lg w-full">
            <Text className="text-white text-base" style={exo2.regular}>
              You will be able to create an account using a registered email and
              valid Tenant ID
            </Text>
          </View>
        </View>

        <View className="flex my-10">
          <View className="flex flex-row gap-2 border-b w-full pb-2 mb-2">
            <Image
              source={require("../assets/icons/mail.png")}
              className="h-5 w-5"
              resizeMode="contain"
            />
            <TextInput
              className="w-full"
              placeholder="Registered email address/Tenant ID"
              keyboardType="email-address"
              onChangeText={(newText) => setEmail(newText)}
              returnKeyType="done"
              textContentType="emailAddress"
              autoCapitalize="none"
              autoComplete="email"
              style={exo2.regular}
            />
          </View>
          {error.length > 0 && (
            <Text style={exo2.regular} className={`${error && "text-red-700"}`}>
              {error}
            </Text> // Display error message if there's an error
          )}
        </View>

        <View className="flex items-center my-10">
          <TouchableOpacity
            onPress={handleSubmit}
            className="bg-[#B80238] w-full items-center flex p-3 rounded-lg"
          >
            <Text
              style={exo2.regular}
              className="text-white uppercase font-medium"
            >
              Next
            </Text>
          </TouchableOpacity>
        </View>

        <View className="relative flex items-center">
          <View className="bg-white rounded-full h-12 w-12 absolute z-30 -top-2 mx-auto flex items-center">
            <Text style={exo2.regular} className="text-black/50">
              OR
            </Text>
          </View>
          <View className="border-black/20 border w-full border-solid "></View>
        </View>

        <View className="flex items-center my-14">
          <View>
            <Text style={exo2.regular} className="text-center font-medium">
              Already have an account
            </Text>
          </View>

          <View className="mt-7">
            <TouchableOpacity
              onPress={() => navigation.navigate("CreateAccount")} // Navigate to "CreateAccount" screen
              className="border-b border-solid pb-1 border-black"
            >
              <Text style={exo2.regular} className="font-medium">
                Log in here
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
}
