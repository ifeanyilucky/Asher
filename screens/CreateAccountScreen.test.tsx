import * as React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import CreateAccountScreen from "./CreateAccountScreen";
import { NavigationContainer } from "@react-navigation/native";

// Mock object
const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

describe("CreateAccountScreen", () => {
  it("shows an error message if email is not provided", () => {
    const { getByText, getByPlaceholderText } = render(
      <NavigationContainer>
        <CreateAccountScreen navigation={navigation} />
      </NavigationContainer>
    );

    const nextButton = getByText("Next");

    // Simulate pressing the "Next" button without entering an email
    fireEvent.press(nextButton);

    // Check if the error message is displayed
    expect(getByText("Email is required.")).toBeTruthy();
  });

  it("shows an error message if email is invalid", () => {
    const { getByText, getByPlaceholderText } = render(
      <NavigationContainer>
        <CreateAccountScreen navigation={navigation} />
      </NavigationContainer>
    );

    const emailInput = getByPlaceholderText(
      "Registered email address/Tenant ID"
    );
    const nextButton = getByText("Next");

    // Simulate user input
    fireEvent.changeText(emailInput, "invalid-email");
    fireEvent.press(nextButton);

    // Check if the error message is displayed
    expect(getByText("Email is invalid.")).toBeTruthy();
  });

  it("navigates to ConfirmPassword if email is valid", () => {
    const { getByText, getByPlaceholderText } = render(
      <NavigationContainer>
        <CreateAccountScreen navigation={navigation} />
      </NavigationContainer>
    );

    const emailInput = getByPlaceholderText(
      "Registered email address/Tenant ID"
    );
    const nextButton = getByText("Next");

    // Simulate user input
    fireEvent.changeText(emailInput, "valid@example.com");
    fireEvent.press(nextButton);

    // Check if the navigation function was called with the expected argument
    expect(navigation.navigate).toHaveBeenCalledWith("ConfirmPassword");
  });

  it("goes back when the back button is pressed", () => {
    const { getByTestId } = render(
      <NavigationContainer>
        <CreateAccountScreen navigation={navigation} />
      </NavigationContainer>
    );

    const backButton = getByTestId("back-button");

    // Simulate pressing the back button
    fireEvent.press(backButton);

    // Check if the goBack function was called
    expect(navigation.goBack).toHaveBeenCalled();
  });
});
