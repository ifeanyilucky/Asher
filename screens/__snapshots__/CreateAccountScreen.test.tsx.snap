// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccountScreen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      fullScreenSwipeShadowEnabled={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              [
                {
                  "backgroundColor": "#fff",
                },
                {
                  "height": 1334,
                },
                {
                  "display": "flex",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginTop": 8,
                  },
                  {
                    "marginBottom": 12,
                  },
                  {
                    "marginLeft": 32,
                    "marginRight": 32,
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                    "width": 48,
                  }
                }
              >
                <View
                  style={
                    {
                      "overflow": "hidden",
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "backgroundColor": "#fff",
                  },
                  {
                    "marginLeft": 40,
                    "marginRight": 40,
                  },
                  {
                    "height": 1334,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "justifyContent": "center",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "marginBottom": 40,
                      "marginTop": 40,
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 36,
                        "lineHeight": 40,
                      },
                      {
                        "fontWeight": "700",
                      },
                      {
                        "color": "#000",
                      },
                      {
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  CREATE AN ACCOUNT
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "marginBottom": 40,
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "#B80238",
                      },
                      {
                        "paddingBottom": 12,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 12,
                      },
                      {
                        "borderBottomLeftRadius": 8,
                        "borderBottomRightRadius": 8,
                        "borderTopLeftRadius": 8,
                        "borderTopRightRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#fff",
                        },
                        {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                      ]
                    }
                  >
                    You will be able to create an account using a registered email and valid Tenant ID
                  </Text>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "marginBottom": 40,
                      "marginTop": 40,
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "display": "flex",
                      },
                      {
                        "flexDirection": "row",
                      },
                      {
                        "marginLeft": -8,
                        "marginTop": -8,
                      },
                      {
                        "borderBottomWidth": 1,
                      },
                      {
                        "width": "100%",
                      },
                      {
                        "paddingBottom": 8,
                      },
                      {
                        "marginBottom": 8,
                      },
                    ]
                  }
                >
                  <Image
                    resizeMode="contain"
                    source={1}
                    style={
                      [
                        {
                          "height": 20,
                        },
                        {
                          "width": 20,
                        },
                        [
                          {
                            "marginLeft": 8,
                            "marginTop": 8,
                          },
                        ],
                      ]
                    }
                  />
                  <TextInput
                    autoCapitalize="none"
                    autoComplete="email"
                    keyboardType="email-address"
                    onChangeText={[Function]}
                    placeholder="Registered email address/Tenant ID"
                    returnKeyType="done"
                    style={
                      [
                        {
                          "width": "100%",
                        },
                        [
                          {
                            "marginLeft": 8,
                            "marginTop": 8,
                          },
                        ],
                      ]
                    }
                    textContentType="emailAddress"
                  />
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "marginBottom": 40,
                      "marginTop": 40,
                    },
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#B80238",
                      "borderBottomLeftRadius": 8,
                      "borderBottomRightRadius": 8,
                      "borderTopLeftRadius": 8,
                      "borderTopRightRadius": 8,
                      "display": "flex",
                      "opacity": 1,
                      "paddingBottom": 12,
                      "paddingLeft": 12,
                      "paddingRight": 12,
                      "paddingTop": 12,
                      "width": "100%",
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#fff",
                        },
                        {
                          "textTransform": "uppercase",
                        },
                        {
                          "fontWeight": "500",
                        },
                      ]
                    }
                  >
                    Next
                  </Text>
                </View>
              </View>
              <View
                style={
                  [
                    {
                      "position": "relative",
                    },
                    {
                      "display": "flex",
                    },
                    {
                      "alignItems": "center",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "#fff",
                      },
                      {
                        "borderBottomLeftRadius": 9999,
                        "borderBottomRightRadius": 9999,
                        "borderTopLeftRadius": 9999,
                        "borderTopRightRadius": 9999,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "width": 48,
                      },
                      {
                        "position": "absolute",
                      },
                      {
                        "zIndex": 30,
                      },
                      {
                        "top": -8,
                      },
                      {
                        "marginLeft": "auto",
                        "marginRight": "auto",
                      },
                      {
                        "display": "flex",
                      },
                      {
                        "alignItems": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "rgba(0, 0, 0, 0.5)",
                        },
                      ]
                    }
                  >
                    OR
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "borderBottomColor": "rgba(0, 0, 0, 0.2)",
                        "borderLeftColor": "rgba(0, 0, 0, 0.2)",
                        "borderRightColor": "rgba(0, 0, 0, 0.2)",
                        "borderTopColor": "rgba(0, 0, 0, 0.2)",
                      },
                      {
                        "borderBottomWidth": 1,
                        "borderLeftWidth": 1,
                        "borderRightWidth": 1,
                        "borderTopWidth": 1,
                      },
                      {
                        "width": "100%",
                      },
                      {
                        "borderStyle": "solid",
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  [
                    {
                      "display": "flex",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "marginBottom": 56,
                      "marginTop": 56,
                    },
                  ]
                }
              >
                <View>
                  <Text
                    style={
                      [
                        {
                          "textAlign": "center",
                        },
                        {
                          "fontWeight": "500",
                        },
                      ]
                    }
                  >
                    Already have an account
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "marginTop": 28,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "borderBottomColor": "#000",
                        "borderBottomWidth": 1,
                        "borderLeftColor": "#000",
                        "borderRightColor": "#000",
                        "borderStyle": "solid",
                        "borderTopColor": "#000",
                        "opacity": 1,
                        "paddingBottom": 4,
                      }
                    }
                  >
                    <Text
                      style={
                        [
                          {
                            "fontWeight": "500",
                          },
                        ]
                      }
                    >
                      Log in here
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="CreateAccount"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
